### PR TITLE
DM-20169: Ensure that fix_header is called following readMetadata

### PR DESCRIPTION
**Pull request by timj**

### CODE REVIEW

File: python/lsst/obs/base/cameraMapper.py [hunk 1]
```diff
@@ -268,6 +272,12 @@ def __init__(self, policy, repositoryDir,
 
         self.makeRawVisitInfo = self.MakeRawVisitInfoClass(log=self.log)
 
+        # Assign a metadata translator if one has not been defined by
+        # subclass. We can sometimes infer one from the RawVisitInfo
+        # class
```
Comment (parejkoj) [inline python/lsst/obs/base/cameraMapper.py]: I fee like this attempt to infer functionality should be documented clearly in the class docstring. Also, last sentence missing a period.

Related to the class docstring, I just noticed that `__init__` has the parameters docstring, not the class: please fix that.

File: python/lsst/obs/base/cameraMapper.py [hunk 2]
```diff
@@ -395,8 +405,12 @@ def setMethods(suffix, mapImpl=None, bypassImpl=None, queryImpl=None):
                                [os.path.join(location.getStorage().root, p) for p in location.getLocations()])
                     # Metadata from FITS file
                     if subPolicy["storage"] == "FitsStorage":  # a FITS image
-                        setMethods("md", bypassImpl=lambda datasetType, pythonType, location, dataId:
-                                   readMetadata(location.getLocationsWithRoot()[0]))
+                        def getMetadata(datasetType, pythonType, location, dataId):
```
Comment (parejkoj) [inline python/lsst/obs/base/cameraMapper.py]: I think this inline function needs a docstring. And the only argument that is used is `location`?

Similarly for the `getBBox`, `getMetadata`, and `getLen` below.

Comment (timj) [reply python/lsst/obs/base/cameraMapper.py]: Aren't they standard butler gen2 callbacks? I don't know what the args are but I assume butler gen2 requires they be there.

Comment (parejkoj) [reply python/lsst/obs/base/cameraMapper.py]: I have no idea. Given the nature of gen2 documentation, I guess we can skip it?

**Pull request merged**